Apply strict mode on moduleMocker.js

fabiomcosta opened the pull request:
I mistakenly said that we couldn't use strict mode on this module.
Adding 'use strict' to it and removing the comment that explained why it wasn't used.

jeffmo commented:
I have to run to a meeting so I haven't looked deeply myself -- but just wanted to point at this code and make sure you'd verified that there's nothing going on in there that can't happen in strict mode:

https://github.com/facebook/jest/blob/master/src/lib/moduleMocker.js#L77

(strict mode has all sorts of things to forget about and I think we're doing some pretty meta stuff in that mockConstructor...and it's been a little while since I added it, but I'd be surprised if I set `/*jshint strict: false*/` without good reason...)

fabiomcosta commented:
@jeffmo, the reason was that `arguments.callee` was used there, but it's use was removed on https://github.com/facebook/jest/commit/739548eb25979788cee7fcdfd339e911644fa224

cpojer commented:
@fabiomcosta is the comment about why it couldn't be used was wrong? We are still using the function constructor in this file.

fabiomcosta commented:
@cpojer, we removed the use of `arguments.callee` and added a function constructor call and changed the comment, incorrectly, saying that we couldn't use 'strict mode' because of the function constructor, but it's ok to use the function constructor on 'strict mode', so I removed the comment and added 'strict mode'.
This is the diff previous to this one that should clarify things https://github.com/facebook/jest/commit/739548eb25979788cee7fcdfd339e911644fa224
